opsdroid/database/matrix/__init__.py

```diff
@@ -0,0 +1,260 @@
+
+        if {**data, **value} == data:
+            _LOGGER.error("Not updating matrix state, as content hasn't changed.")
+            return
+        else:
+            data = {**data, **value}
+
+        if self.is_room_encrypted and self.should_encrypt:
+            for k, v in value.items():
+                room_event = await self.opsdroid.send(
+                    GenericMatrixRoomEvent(
+                        target=self.room_id, content={k: v}, event_type=self._event_type
+                    )
+                )
+                data[k] = {"encrypted_val": room_event.event_id}
+
+        _LOGGER.debug(f"Putting {key} into matrix room {self.room_id} with {data}")
+
+        await self.opsdroid.send(
+            MatrixStateEvent(
+                self._event_type,
+                content=data,
+                target=self.room_id,
+                connector=self.connector,
+                state_key=state_key,
+            )
+        )
+
+        return True
+
+    async def get(self, key, check_dict=None):
+        """Get a value from the database for a given key."""
+
+        if self.should_migrate:
+            await self.migrate_database()
+
+        # If the single state key flag is set then use that else use state key.
+        state_key = (
+            "" if self._single_state_key is True else self._single_state_key or key
+        )
+
+        _LOGGER.debug(
+            f"Getting {key} from matrix room {self.room_id} with state_key={state_key}."
+        )
+
+        data = await self.connector.connection.room_get_state_event(
+            room_id=self.room_id, event_type=self._event_type, state_key=state_key,
+        )
+        if isinstance(data, RoomGetStateEventError):
+            _LOGGER.error(
+                f"Error getting {key} from matrix room {self.room_id}: {data.message}({data.status_code})"
+            )
+            return
+
+        if data.transport_response.status == 404:
+            data.content.clear()
+        data = data.content
+
+        _LOGGER.debug(f"Got {data} from state request.")
+
+        check = check_dict or data
+
+        for k, v in data.items():
+            if isinstance(v, dict) and "encrypted_val" in v and k in check:
+                resp = await self.connector.connection.room_get_event(
+                    room_id=self.room_id, event_id=v["encrypted_val"],
+                )
+                if isinstance(resp, RoomGetEventError):
+                    _LOGGER.error(
+                        f"Error decrypting {v['encrypted_val']} while getting "
+                        f"{key}: {resp.message}({resp.status_code})"
+                    )
+                    continue
+                data[k] = resp.event.source["content"][k]
+
+        try:
+            if self._single_state_key and not check_dict:
+                data = data[key]
+        except KeyError:
+            _LOGGER.debug(f"{key} doesn't exist in database")
+            return None
+
+        return data
```

Comment:
I think with my suggestion above this can just be replaced with:
```suggestion
        if self._single_state_key:
            return data.get(key)
        return data
```